app/models/form/question.rb

```diff
@@ -271,7 +268,7 @@ def selected_answer_option_is_derived?(log)
   end
 
   def has_inferred_display_value?(log)
-    inferred_check_answers_value.present? && log[inferred_check_answers_value["condition"].keys.first] == inferred_check_answers_value["condition"].values.first
+    inferred_check_answers_value.present? && inferred_check_answers_value.any? { |inferred_value| log[inferred_value["condition"].keys.first] == inferred_value["condition"].values.first }
```

Comment:
Why have you used `.first` here but `[0]` in `inferred_check_answers_value` and `inferred_answer_value`?

Reply:
No reason, it does the same thing,  I can change the ones in `inferred_answer_value` if that's more readable

Reply:
Not sure which bit you meant by `inferred_check_answers_value` tho?

Reply:
Sorry that was a typo, I meant `has_inferred_check_answers_value?`